src/Access/Common/AuthenticationData.h

```diff
@@ -32,6 +32,8 @@ enum class AuthenticationType
     /// Certificates may only be trusted if 'strict' SSL mode is enabled.
     SSL_CERTIFICATE,
 
+    INTERSERVER_SECRET,
```

Comment:
This doesn't seem to be correct. The idea of the authentication type is that you can specify it when you create a user and then it's used while he logins. It doesn't make sense to create a user with `INTERSERVER_SECRET` as its authentication type, it's just a mode inside TCP protocol.

Reply:
I think we had tests for interserver secret I wonder why they didn't fail...

Reply:
The reason why I added it here is that it's also incorrect to write `NO_PASSWORD` into `auth_type` column of session_log when authentication was done through "interserver secret" (especially, when user's auth type is different from `NO_PASSWORD`). It seemed to me that "interserver secret" is just a special authentication type because we pass special `AlwaysAllowCredentials` to `AccessControl::authenticate` and `Authentication::areCredentialsValid` (just like we do for ordinary auth types). And actually it is, but it authenticates a server, not a user (so `AlwaysAllowCredentials` does not really make sense too). I think I will remove `INTERSERVER_SECRET` from `AuthenticationType` and will simply make `auth_type` column `Nullable`.